head/firmware/motor_hardware.c

```diff
@@ -181,6 +184,8 @@ void MX_GPIO_Init(void) {
     __HAL_RCC_GPIOF_CLK_ENABLE();
     __HAL_RCC_GPIOA_CLK_ENABLE();
     __HAL_RCC_GPIOB_CLK_ENABLE();
+    __HAL_RCC_GPIOD_CLK_ENABLE();
```

Comment:
what do we use portd for?

Reply:
we need port D for ENCI_Z_Axis pin. That's the only pin on port D. That's why i enabled the CLK on this port.